huaweicloud/services/dli/resource_huaweicloud_dli_table.go

```diff
@@ -296,7 +326,20 @@ func setStoragePropertiesToState(d *schema.ResourceData, storageProperties []map
 	return mErr.ErrorOrNil()
 }
 
-func resourceDliTableDelete(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func datatableDeleteRefreshFunc(client *golangsdk.ServiceClient, dbName, tableName string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		detail, queryErr := tables.Get(client, dbName, tableName)
+		if queryErr != nil {
+			if _, ok := queryErr.(golangsdk.ErrDefault404); !ok {
+				return detail, "ERROR", queryErr
+			}
+			return detail, "COMPLETED", nil
```

Comment:
Variable `detail` may be nil.

Reply:
the result of Get() method during golangsdk will have the type value (corresponding structure) in any times, will not be nil (nil means the type and value are both empty)